### ISSUE TITLE
SA1501 complains about empty catch block

### ISSUE THREAD

**Issue by 3geek14** [DotNetAnalyzers/StyleCopAnalyzers#3564]
My understanding of SA1501 is that it objects to a single statement in a code block being on the same line as the opening and closing braces, like this:

    if (condition) { MethodCall(); } // invalid

For empty code blocks, I'm used to SA1501 allowing it, like this:

    public MyException() : base() { } // valid

I reecntly ran into an empty code block that SA1501 objected to, and I think it's a bug:

    try {
      MethodCall();
    } catch { } // invalid

I know that empty catch blocks are generally frowned upon. I think it would be reasonable for StyleCop to have a (separate) analyzer looking for that, but SA1501 is not currently doing that. Notably, adding a line break satisfies SA1501 while still leaving an empty catch block:

    try {
      MethodCall();
    } catch {
    } // valid

Is this a bug? If it's not a bug, I think the documentation for SA1501 should be updated to elaborate.

**Comment by bjornhellander** [comment]
Hi!

In my mind, frowning about empty catch blocks is done because of behaviour and not style, so I don't think StyleCop.Analyzers would be the place for that kind of rule.

SA1501 concerns **statements** with braces (if, while, for, lock, etc). Since the braces in your constructor example is not part of a statement, it does not fire. SA1502 (which concerns **elements** with braces) does however fire, so I assume that you have turned that one off, since you don't see it.

To me, the behaviour you are getting seems to be as designed.

**Comment by 3geek14** [comment]
In that case, I think the documentation is confusing:

> A violation of this rule occurs when **a statement that is wrapped in opening and closing braces** is written on a single line.

In my case, nothing is wrapped in opening and closing braces. If I instead had `} catch { ; }` as the final line, there would be an empty statement wrapped in opening and closing braces, but I have not even an empty statement.

Perhaps a better wording would be this?

> A violation of this rule occurs when the body of a statement is written with its opening and closing braces on the same line.

**Comment by bjornhellander** [comment]
Yes, I also think the the original test is misleading, but I am unsure of how to phrase it. Some other documentation pages mentions _statement blocks_, so this could also be an alternative:
> A violation of this rule occurs when a statement block is written with its opening and closing braces on the same line.

Would you accept a pull request to clarify SA1501 with a change in line with any of these two suggestions, @sharwell? Any opinion on the actual wording?
